src/serac/infrastructure/profiling.hpp

```diff
@@ -111,21 +120,53 @@ auto&& forwarder(T&& thing)
   return std::forward<T>(thing);
 }
 
-}  // namespace detail
+  /**
+   * @brief Guarantees str is a c string
+   */
+template <typename T>
+ std::enable_if_t<std::is_same_v<const char *, T>, const char * > make_cstr(T str) {
+    return str;
+}
+
+  /**
+   * @brief Converts a std::string into a c string
+   */  
+template <typename T>
+std::enable_if_t<std::is_same_v<std::string, T>, const char * > make_cstr(const T & str) {
+    return str.c_str();
+}
```

Comment:
Would just an overload be sufficient here?
```cpp
const char* make_cstr(const char*);
const char* make_cstr(const std::string&);
```

Reply:
Funny. I swore I tried this in godbolt yesterday and it claimed it was ambiguous.